### PR TITLE
Revamping User Update step with aggregating examples

### PR DESCRIPTION
**Pull request by josh-mccrowell-braze**

### CODE REVIEW

File: _docs/_user_guide/engagement_tools/canvas/canvas_components/user_update.md [hunk 1]
```diff
@@ -28,19 +28,71 @@ There are three options that allow you to update existing, add new, or remove us
...
 
-### Advanced JSON composer
+### Updating arrays
+
+The user update step can add or remove attributes to an [array of objects]({{site.baseurl}}/user_guide/data_and_analytics/custom_data/custom_attributes/array_of_objects/). Select an array attribute name from your list of attributes and enter the key value.
+
+#### Example: Updating the user's wishlist
+
+By adding or removing an item to an array, you can update the user's wishlist. <!--Note for Josh - want to resize the example images -->
+
```
Comment (johndinh) [inline _docs/_user_guide/engagement_tools/canvas/canvas_components/user_update.md:50]: Are we adding sample code here?

File: _docs/_user_guide/engagement_tools/canvas/canvas_components/user_update.md [hunk 2]
```diff
@@ -28,19 +28,71 @@ There are three options that allow you to update existing, add new, or remove us
...
+![][9]{: style="max-width:90%;"}
+
+### Increasing and decreasing values
+
+The user update step can increase or decrease an attribute value. Select the attribute, select **Increment By** or **Decrement By**, and enter a number.
```
Comment (johndinh) [inline _docs/_user_guide/engagement_tools/canvas/canvas_components/user_update.md:55]: Need an example but I am pretty sure you can do this use case with the UI editor.  Don't need advanced JSON.

File: _docs/_user_guide/engagement_tools/canvas/canvas_components/user_update.md [hunk 3]
```diff
@@ -28,19 +28,71 @@ There are three options that allow you to update existing, add new, or remove us
...
+
+![][9]{: style="max-width:90%;"}
+
+### Increasing and decreasing values
+
+The user update step can increase or decrease an attribute value. Select the attribute, select **Increment By** or **Decrement By**, and enter a number. 
+
+#### Example: Track weekly progress
+
+By incrementing a custom attribute that tracks an event, you could track the number of classes that a user has taken in a week. Using this component, the class count can reset at the start of the week and begin tracking again.
```
Comment (johndinh) [inline _docs/_user_guide/engagement_tools/canvas/canvas_components/user_update.md:59]: Similarly, this is an increment use case that can be handled in the UI editor.

File: _docs/_user_guide/engagement_tools/canvas/canvas_components/user_update.md [hunk 4]
```diff
@@ -28,19 +28,71 @@ There are three options that allow you to update existing, add new, or remove us
...
+
+![][7]{: style="max-width:90%;"}
+
+### Aggregating attributes
+
+The user update step can calculate fields and aggregate user data. <!--The exact steps and best practices to do this TBD-->
+
+#### Example: Calculate shopping cart total
+
+<!--TBD-->
```
Comment (johndinh) [inline _docs/_user_guide/engagement_tools/canvas/canvas_components/user_update.md:69]: Presumably the example is that we iterate through an array of objects (the cart) and then accumulate the total cost of the cart and store as an attribute?

File: _docs/_user_guide/engagement_tools/canvas/canvas_components/user_update.md [hunk 5]
```diff
@@ -28,19 +28,71 @@ There are three options that allow you to update existing, add new, or remove us
...
+
+#### Example: Calculate user's total lifetime value
+
+<!--TBD-->
```
Comment (johndinh) [inline _docs/_user_guide/engagement_tools/canvas/canvas_components/user_update.md:73]: Presumably we will add a cart value to an existing LTV?

File: _docs/_user_guide/engagement_tools/canvas/canvas_components/user_update.md (hunk 1 above)
Comment (josh-mccrowell-braze) [reply _docs/_user_guide/engagement_tools/canvas/canvas_components/user_update.md:50]: I wasn't planning on using code. The attached image shows how an attribute added to an array of attributes can do something like updating a wishlist. If there's a better example to use, happy to use it!

File: _docs/_user_guide/engagement_tools/canvas/canvas_components/user_update.md (hunk 5 above)
Comment (johndinh) [reply _docs/_user_guide/engagement_tools/canvas/canvas_components/user_update.md:73]: `{
  "attributes": [
    {
      "ltv": {{ custom_attribute.${ltv} | plus: canvas_entry_properties.${order_total} }}
    }
  ]
}`

In this example, the order total is pull from the canvas entry property but customers can template in their own liquid.

File: _docs/_user_guide/engagement_tools/canvas/canvas_components/user_update.md (hunk 4 above)
Comment (johndinh) [reply _docs/_user_guide/engagement_tools/canvas/canvas_components/user_update.md:69]: I'd ask rebecca for an example here...

**Review by lydia-xie — APPROVED** [review]
lgtm!

**Pull request merged**